src/CRISP.BackendChallenge/Controllers/EmployeeController.cs

```diff
@@ -7,33 +7,266 @@
-    private readonly IRepository<Employee> _repository;
+    private readonly IRepository<Employee> _employeeRepository;
+    private readonly IRepository<Login> _loginRepository;
 
-    public EmployeeController(ILogger<EmployeeController> logger, IRepository<Employee> repository)
+    public EmployeeController(ILogger<EmployeeController> logger, IRepository<Employee> employeeRepository, IRepository<Login> loginRepository)
     {
         _logger = logger;
-        _repository = repository;
+        _employeeRepository = employeeRepository;
+        _loginRepository = loginRepository;
     }
 
     [HttpGet]
+    [Route("all")]
     public IActionResult GetAll()
     {
         _logger.LogDebug(":: Performing {MethodName}", nameof(GetAll));
-        var result = _repository.Query()
-            .ToList().Select(x => new EmployeeResponse
+            
+        try
+        {
+            var employees = _employeeRepository.GetAll();
+
+            var result = employees.Select(ToEmployeeResponse);
+
+            return Ok(result);
+        }
+        catch (ArgumentNullException e)
+        {
+            _logger.LogDebug(":: There are no Employees in the Database: {EMessage}", e.Message);
+            return NotFound(new {error = "No Employees Found in Database"});
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(":: Error occurred while getting all employees: {EMessage}", e.Message);
+            return CreateInternalServerError();
+        }
+    }
+
+    [HttpGet]
+    [Route("{id:int}")]
+    public IActionResult GetById(int id)
+    {
+        _logger.LogDebug(":: Performing {MethodName} with ID {Id}", nameof(GetById), id);
+        try
+        {
+            var employee = _employeeRepository.GetById(id);
+            return Ok(ToEmployeeResponse(employee));
+        }
+        catch (ArgumentNullException e)
+        {
+            _logger.LogDebug(":: Id Not Found: {Id}, {EMessage}", id, e.Message);
+            return CreateNotFound(id);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(":: Error occurred while getting employee ID: {Id} {EMessage}", id, e.Message);
+            return CreateInternalServerError();
+        }
+    }
+    
+    
+    [HttpPost]
+    [Route("createEmployee")]
+    public IActionResult AddEmployee([FromBody] EmployeeRequest employeeRequest)
+    {
+        _logger.LogDebug(":: Performing {MethodName}", nameof(AddEmployee));
+        
+        if (string.IsNullOrWhiteSpace(employeeRequest.Name))
+        {
+            _logger.LogError(":: Invalid EmployeeRequest provided");
+            return CreateBadRequest("Creating an employee", "Name field required");
+        }
+        if (employeeRequest.Department == null)
+        {
+            _logger.LogError(":: Invalid EmployeeRequest provided");
+            return CreateBadRequest("Creating an employee", "Department field required");
+        }
```

Comment:
Request validation logic is better performed by leveraging the ModelState functionality provided by the framework. There are many ways to do this, but for this simple case I would suggest data attributes:

```cs

public class EmployeeRequest
{
    [Required]
    public string? Name { get; set; }

    [Required]
    public int? Department { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public List<DateTime>? LoginDates { get; set; }
}

... 
    public IActionResult AddEmployee([FromBody] EmployeeRequest employeeRequest)
    {
      if (!ModelState.IsValid) return CreateBadRequest(ModelState);
    }

    public static BadRequestObjectResult CreateBadRequest(ModelState modelState)
    {
      // Do something with the modelstate to return a useful error. It will contain all of the info you need
    }
```

Reply:
Created a ModelState Filter to handle incoming invalid requests